### DISCUSSION TITLE
Is there a way to use the patterns built in to ImageMagick

### DISCUSSION THREAD

**Discussion by cstrachan88** [dlemstra/Magick.NET#654] [Help] [answered]
In the past, I have used the command line to overlay patterns on images using the -size command:

* `magick -size 64x64 pattern:RIGHT45 output.png`
* `magick img.png ( -size 64x64 pattern:RIGHT45 -transparent white ) -flatten output.png`

You can find [documentation for this here](https://imagemagick.org/script/formats.php#builtin-patterns). Is there a way to access this through Magick.NET? Poking around, I don't see a way to do that currently in the documentation or through code.

See also issue #3 from a few years back.

**Comment by dlemstra** [comment, marked as answer]
You can use the pattern as the filename in the constructor of a `MagickImage`:

```C#
using (var image = new MagickImage("pattern:RIGHT45", 64, 64))
{
    image.Write("output.png");
}
```

And for the second command you will need to use a `MagickImageCollection`:

```C#
using (var images = new MagickImageCollection())
{
    images.Add(new MagickImage("img.png"));

    var right = new MagickImage("pattern:RIGHT45", 64, 64);
    right.Transparent(MagickColors.White);
    images.Add(right);

    using (var output = images.Flatten())
    {
        output.Write("output.png");
    }
}
```
